src/test/java/com/linecorp/armeria/client/http/HttpClientIntegrationTest.java

```diff
@@ -255,9 +273,45 @@ public void testDoubleSlashSuppression() throws Exception {
 
     private static void testDoubleSlashSuppression(String path, String normalizedPath) throws IOException {
         testSocketOutput(path,
-                         port -> "GET " + normalizedPath + " HTTP/1.1\r\n" +
-                                 "user-agent: Armeria\r\n" +
-                                 "host: 127.0.0.1:" + port + "\r\n\r\n");
+                port -> "GET " + normalizedPath + " HTTP/1.1\r\n" +
+                        "host: 127.0.0.1:" + port + "\r\n" +
+                        "user-agent: Armeria\r\n\r\n"
+        );
+    }
+
+    /**
+     * User-agent header should be overridden by ClientOption.HTTP_HEADER
+     */
+    @Test
+    public void testUserAgentOverridableByClientOption() throws Exception {
+
+        HttpHeaders headers = HttpHeaders.of(HttpHeaderNames.USER_AGENT, "ArmeriaTest");
+        ClientOptions options = ClientOptions.of(ClientOption.HTTP_HEADERS.newValue(headers));
+
+        HttpClient client = Clients.newClient(clientFactory, "none+http://127.0.0.1:" + httpPort,
+                HttpClient.class, options);
+
+        AggregatedHttpMessage response = client.execute(
+                HttpHeaders.of(HttpMethod.GET, "/useragent")).aggregate().get();
+
+        assertEquals("ArmeriaTest", response.content().toString(StandardCharsets.UTF_8));
+    }
+
+    @Test
+    public void testUserAgentOverridableByRequestHeader() throws Exception {
+
+        HttpHeaders headers = HttpHeaders.of(HttpHeaderNames.USER_AGENT, "ArmeriaTest");
+        ClientOptions options = ClientOptions.of(ClientOption.HTTP_HEADERS.newValue(headers));
+
+        HttpClient client = Clients.newClient(clientFactory, "none+http://127.0.0.1:" + httpPort,
+                HttpClient.class, options);
+
+        AggregatedHttpMessage response = client.execute(
+                HttpHeaders.of(HttpMethod.GET, "/useragent")
+                        .add(HttpHeaderNames.USER_AGENT, "Overidden"))
```

Comment:
Extract `Overridden` into a variable so that the `assertEquals` below does not duplicate it.